File: camerax_ios/ios/Classes/ZoomState.swift
//
//  ZoomState.swift
//  camerax_ios
//
//  Created by 闫守旺 on 2024/7/12.
//

import Foundation

@objc public class ZoomState: NSObject {
    @objc public let minZoomRatio: Double
    @objc public let maxZoomRatio: Double
    @objc public let zoomRatio: Double
    
    init(minZoomRatio: Double, maxZoomRatio: Double, zoomRatio: Double) {
        self.minZoomRatio = minZoomRatio
        self.maxZoomRatio = maxZoomRatio
        self.zoomRatio = zoomRatio
    }
    
    @objc public var linearZoom: Double {
        (zoomRatio - minZoomRatio) / (maxZoomRatio - minZoomRatio)
    }
}

File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/core/ZoomState.kt
package dev.hebei.camerax_android.core

class ZoomState internal constructor(internal val obj: androidx.camera.core.ZoomState) {
    val minZoomRatio get() = obj.minZoomRatio
    val maxZoomRatio get() = obj.maxZoomRatio
    val zoomRatio get() = obj.zoomRatio
    val linearZoom get() = obj.linearZoom
}